build_tools/buildkite/dana/upload_benchmarks.py

```diff
@@ -0,0 +1,297 @@
+IREE_PROJECT_ID = 'IREE'
+
+# Number of total retries when post to dashboard.
+RETRY_COUNT = 5
+
+# A non-exhaustive list of models and their descriptions.
+# For models listed here we can provide a nicer description for them on
+# webpage.
+IREE_MODELS = {
+    'MobileNetV2 (imagenet) [TensorFlow]':
+        'MobileNet v2 from https://www.tensorflow.org/api_docs/python/tf/keras/applications/MobileNetV2',
+    'MobileNetV3Small (imagenet) [TensorFlow]':
+        'MobileNet v3 from https://www.tensorflow.org/api_docs/python/tf/keras/applications/MobileNetV3Small',
+    'MobileBertSquad [TensorFlow]':
+        'MobileBERT (SQuAD) from https://github.com/google-research/google-research/tree/master/mobilebert'
+}
+
+# A non-exhaustive list of phones and their descriptions.
+# For phones listed here we can provide a nicer description for them on
+# webpage.
+IREE_PHONES = {
+    'Pixel-4':
+        'Google Pixel 4 chipset: Snapdragon 855 (https://en.wikichip.org/wiki/qualcomm/snapdragon_800/855)',
+    'SM-G980F':
+        'Samsung Galaxy S20 chipset: Exynos 990 (https://en.wikichip.org/wiki/samsung/exynos/990)',
+}
+
+
+def execute(args,
+            cwd,
+            capture_output=False,
+            treat_io_as_text=True,
+            verbose=False,
+            **kwargs):
+  """Executes a command."""
+  if verbose:
+    print(f"+{' '.join(args)}  [from {cwd}]")
+  return subprocess.run(args,
+                        cwd=cwd,
+                        check=True,
+                        capture_output=capture_output,
+                        text=treat_io_as_text,
+                        **kwargs)
+
+
+def get_git_commit_hash(commit, verbose=False):
+  """Gets the commit hash for the given commit."""
+  return execute(['git', 'rev-parse', commit],
+                 cwd=os.path.dirname(os.path.realpath(__file__)),
+                 capture_output=True,
+                 verbose=verbose).stdout.strip()
+
+
+def get_git_total_commit_count(commit, verbose=False):
+  """Gets the total commit count in history ending with the given commit."""
+  count = execute(['git', 'rev-list', '--count', commit],
+                  cwd=os.path.dirname(os.path.realpath(__file__)),
+                  capture_output=True,
+                  verbose=verbose).stdout.strip()
+  return int(count)
+
+
+def get_git_commit_info(commit, verbose=False):
+  """Gets commit information dictory for the given commit."""
+  info = execute([
+      'git', 'show', '--format=%H:::%h:::%an:::%ae:::%s', '--no-patch', commit
+  ],
+                 cwd=os.path.dirname(os.path.realpath(__file__)),
+                 capture_output=True,
+                 verbose=verbose).stdout.strip()
+  segments = info.split(':::')
+  return {
+      'hash': segments[0],
+      'abbrevHash': segments[1],
+      'authorName': segments[2],
+      'authorEmail': segments[3],
+      'subject': segments[4],
+  }
+
+
+def compose_serie_payload(project_id,
```

Comment:
series? I don't think use of "serie" is really around anymore. You can have "a series" for sure

```suggestion
def compose_series_payload(project_id,
```

Reply:
It was following the name from Dana. But yeah you are right it's not commonly used.